src/lib.rs

```diff
@@ -9,7 +9,8 @@
 
 // From rustc.
 extern crate arena;
-#[macro_use] extern crate rustc;
+#[macro_use]
+extern crate rustc;
```

Comment:
You mean like this?

```rust
#[macro_use]
extern crate rustc;

extern crate arena;
extern crate rustc_data_structures;
extern crate rustc_mir;
extern crate syntax;
```

I agree that putting a newline before attributes makes sense.

However, re-ordering statements isn't rustfmt can reliably do due to comments. That is, there's no way to determine if `// From rustc.` should stick to `extern crate arena;` or if it should stay where it is.